### PR TITLE
`dojo-client` partial world syncer

### PR DESCRIPTION
**Pull request by kariy**
Just a draft. May or may not be merged

- [x] is wasmable?

Adds a type in the `dojo-client` which allows to sync a set of entities' component values.

### CODE REVIEW

File: crates/dojo-client/src/storage.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /// This function mimic `world::entity` of `dojo-core`
+    async fn get(
+        &self,
+        component: FieldElement,
+        keys: Vec<FieldElement>,
+        length: usize,
+    ) -> Vec<FieldElement>;
+}
+
+pub fn get_component_storage_base_address(
```
Comment (tarrencev) [inline crates/dojo-client/src/storage.rs:27]:
```suggestion
pub fn component_storage_base_address(
```

File: crates/dojo-client/src/sync.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        Self { world_reader, storage, entity_components_to_sync }
+    }
+
+    // TODO: wrap this logic in a Future
+    /// start the syncing loop.
+    /// fetch the requested components values for every fixed interval.
+    pub async fn start(&mut self) {
+        let mut entity_readers = Vec::new();
+
+        for i in &self.entity_components_to_sync {
```
Comment (tarrencev) [inline crates/dojo-client/src/sync.rs:58]: In the final impl, we'll want a way to register new entities and components at runtime. For this simple example this is fine though

**Review by tarrencev — APPROVED** [review]
Looking really good!

**Comment by kariy** [comment]
@tarrencev should we keep the client example? i put that just to help making sense and explaining how it would be used.

**Review by tarrencev — APPROVED** [review]

**Pull request merged**